api/src/setup/setup.service.ts

```diff
@@ -1,79 +1,83 @@
-import mongoose, { Model } from 'mongoose';
+import type { Setup } from '@prisma/client';
+import type { Model } from 'mongoose';
 
+import { InjectModel } from '@/core/decorators/inject-prisma-client.decorator';
 import { DatasetsService } from '@/datasets/datasets.service.js';
+import type { CreateTabularDatasetDto } from '@/datasets/zod/dataset';
 import { UsersService } from '@/users/users.service.js';
 
-import { SetupConfig } from './schemas/setup-config.schema.js';
-
-import type { CreateAdminDto, SetupDto } from './dto/setup.dto.js';
+import type { CreateAdminDto, SetupDto } from './zod/setup.js';
 
 @Injectable()
 export class SetupService {
   constructor(
-    @InjectModel(SetupConfig.name) private readonly setupConfigModel: Model<SetupConfig>,
-    @InjectConnection() private readonly connection: mongoose.Connection,
+    @InjectModel('Setup') private readonly setupModel: Model<Setup>,
     private readonly datasetsService: DatasetsService,
     private readonly usersService: UsersService
-  ) {}
+  ) { }
 
   async getSetupConfig() {
-    const setupConfig = await this.setupConfigModel.findOne();
+    const setupConfig = await this.setupModel.findOne();
     if (!setupConfig) {
       throw new NotFoundException('Setup Config not found in the database.');
     }
     return setupConfig;
   }
 
-  async getState(): Promise<SetupState> {
+  async getState() {
     return { isSetup: await this.isSetup() };
   }
 
   async getVerificationInfo() {
-    const verificationInfo = (await this.getSetupConfig()).verificationInfo;
+    const verificationInfo = await this.setupModel.findOne();
     if (!verificationInfo) {
       throw new NotFoundException('Cannot access verification info.');
     }
     return verificationInfo;
   }
 
   async initApp({ admin, setupConfig }: SetupDto) {
-    console.log(setupConfig);
     if (await this.isSetup()) {
       throw new ForbiddenException();
     }
-    await this.connection.dropDatabase();
     const user = await this.createAdmin(admin);
 
-    await this.setupConfigModel.create(setupConfig);
+    await this.setupModel.create({
+      adminId: user.id,
+      userVerification: setupConfig
+    });
 
-    const iris = await this.loadStarterDataset('iris.json');
-    await this.datasetsService.createDataset(iris, user.toObject());
+    const createStarterDatasetDto: CreateTabularDatasetDto = {
+      datasetType: 'TABULAR',
+      description: 'a sample dataset containing data about iris',
+      license: 'PUBLIC',
+      name: 'iris',
+      primaryKeys: []
+    }
+    await this.datasetsService.createDataset(
+      createStarterDatasetDto,
+      await fs.readFile(path.resolve(import.meta.dir, 'resources', 'iris.json'), 'utf-8'),
+      user.id);
   }
 
   private async createAdmin(admin: CreateAdminDto) {
     return this.usersService.createUser({
       ...admin,
-      confirmedAt: Date.now(),
-      role: 'admin',
-      verifiedAt: Date.now()
+      confirmedAt: new Date(Date.now()),
```

Comment:
This is redundant, since calling `new Date()` with no arguments defaults to the value of `Date.now()`